grails-datastore-gorm-hibernate5/src/main/groovy/org/grails/orm/hibernate/dirty/GrailsEntityDirtinessStrategy.groovy

```diff
@@ -56,6 +60,30 @@ class GrailsEntityDirtinessStrategy implements CustomEntityDirtinessStrategy {
     void resetDirty(Object entity, EntityPersister persister, Session session) {
         if (canDirtyCheck(entity, persister, session)) {
             cast(entity).trackChanges()
+            try {
+                PersistentEntity persistentEntity = GormEnhancer.findEntity(Hibernate.getClass(entity))
+                if (persistentEntity != null) {
+                    resetDirtyEmbeddedObjects(persistentEntity, entity, persister, session)
+                }
+            } catch (IllegalStateException e) {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug(e.message, e)
```

Comment:
Why are we swallowing the exception here and logging at `debug` level?

Reply:
With this PR, we are recursively resetting the `$changedProperties` of embedded objects. In cases where the embedded object is not a PersistentEntity the recursion will break, and `IllegalStateException` is thrown as: 
```
Either class [grails.gorm.tests.dirtychecking.Address] is not a domain class or GORM has not been initialized correctly or has already been shutdown. Ensure GORM is loaded and configured correctly before calling any methods on a GORM entity.
```